odoo/client.go

```diff
@@ -6,26 +6,49 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+	"strings"
 	"time"
 )
 
 // Client is the base struct that holds information required to talk to Odoo
 type Client struct {
 	parsedURL *url.URL
 	db        string
+	username  string
+	password  string
 	http      *http.Client
 }
 
 // NewClient returns a new client with its basic fields set.
+// The URL must be in the format of `https://user:pass@host[:port]/db-name`.
 // It returns error if baseURL is not parseable with url.Parse.
-func NewClient(baseURL, db string) (*Client, error) {
+func NewClient(baseURL string) (*Client, error) {
 	u, err := url.Parse(baseURL)
 	if err != nil {
 		return nil, fmt.Errorf("proper URL format is required: %w", err)
 	}
+
+	if u.User == nil || u.User.Username() == "" {
+		return nil, fmt.Errorf("missing username and password in URL")
+	}
+	pw := ""
+	if value, set := u.User.Password(); set {
+		pw = value
+	} else {
+		return nil, fmt.Errorf("missing password in URL")
+	}
+
+	// Technical debt: This means an Odoo running under a path like https://odoo/pathprefix/ can't be parsed.
```

Comment:
I think this is fine. We could potentially just try to take the last segment? Not sure if that's necessary (or intuitive)

Reply:
I'd leave it like this. The more segments the confuser...